controllers/managedmcg_controller.go

```diff
@@ -215,30 +251,87 @@ func (r *ManagedMCGReconciler) reconcilePhases() (reconcile.Result, error) {
 		}
 
 		// Reconcile the different resources
-		if err := r.reconcileODFOperatorMgrConfig(); err != nil {
+		//TODO : remove once ODF upgrad to 4.10
+		/*if err := r.reconcileODFOperatorMgrConfig(); err != nil {
 			return ctrl.Result{}, err
-		}
+		}*/
 		if err := r.reconcileConsoleCluster(); err != nil {
 			return ctrl.Result{}, err
 		}
 		if err := r.reconcileStorageSystem(); err != nil {
 			return ctrl.Result{}, err
 		}
-		if err := r.reconcileODFCSV(); err != nil {
+		//TODO : remove once ODF upgrad to 4.10
+		/*if err := r.reconcileODFCSV(); err != nil {
 			return ctrl.Result{}, err
-		}
+		}*/
 		if err := r.reconcileNoobaa(); err != nil {
 			return ctrl.Result{}, err
 		}
+		//TODO : remove once ODF upgrad to 4.10
 		if err := r.reconcileStorageCluster(); err != nil {
 			return ctrl.Result{}, err
 		}
 		r.managedMCG.Status.ReconcileStrategy = r.reconcileStrategy
 
+		// Check if we need and can uninstall
+		if initiateUninstall && r.areComponentsReadyForUninstall() {
+
+			r.Log.Info("starting MCG uninstallation - deleting managedMCG")
+			if err := r.delete(r.managedMCG); err != nil {
+				return ctrl.Result{}, fmt.Errorf("unable to delete managedMCG: %v", err)
+			}
+
+			if err := r.get(r.managedMCG); err != nil {
+				if !errors.IsNotFound(err) {
+					return ctrl.Result{}, err
+				}
+				r.Log.V(-1).Info("Trying to reload managedMCG resource after delete failed, managedMCG resource not found")
+			}
+		}
+
+	} else if initiateUninstall {
+		return ctrl.Result{}, r.removeOLMComponents()
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagedMCGReconciler) removeOLMComponents() error {
+
+	r.Log.Info("deleting deployer csv")
+	if csv, err := r.getCSVByPrefix(deployerCSVPrefix); err == nil {
+		if err := r.delete(csv); err != nil {
+			return fmt.Errorf("Unable to delete csv: %v", err)
+		} else {
+			r.Log.Info("Deployer csv removed successfully")
+			return nil
+		}
+	} else {
+		return err
+	}
+}
+
+func (r *ManagedMCGReconciler) areComponentsReadyForUninstall() bool {
+	subComponents := r.managedMCG.Status.Components
+	return subComponents.Noobaa.State == mcgv1alpha1.ComponentReady
+}
+
+func (r *ManagedMCGReconciler) checkUninstallCondition() bool {
+	configmap := &corev1.ConfigMap{}
```

Comment:
What are we using this ConfigMap for? Is there anything being stored in it? Do we really need this?

Reply:
@rexagod  This ConfigMap is created from addon when we click on the uninstall button in OCM, and its like signal us to delete the resources, It doesn't have any data but annotations

Reply:
Hmm, okay.